go/pkg/client/bytestream_test.go

```diff
@@ -0,0 +1,294 @@
+			notFound:    false,
+		},
+	}
+
+	b := setup(t)
+	defer b.shutDown()
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			lsID := test.ls.logStreamID
+			if !test.notFound {
+				logStreams[lsID] = test.ls
+			}
+			data := []byte("Hello World!")
+			ChunkMaxSize(len(data)).Apply(b.client)
+			writtenBytes, err := b.client.WriteBytesWithOptions(b.ctx, lsID, data, test.opts...)
+			if err == nil {
+				t.Error("WriteBytesWithOffset() got nil error, want non-nil error")
+			}
+			if writtenBytes != 0 {
+				t.Errorf("WriteBytesWithOffset() got %d byte(s), want 0 byte", writtenBytes)
+			}
+		})
+	}
+}
+
+func TestWirteBytesWithOptions_FinishWrite(t *testing.T) {
+	b := setup(t)
+	defer b.shutDown()
+	dataSize := len(logStreamData)
+	for i, flag := range []ByteStreamWriteOption{ByteSteramOptFinishWrite(true), ByteSteramOptFinishWrite(false)} {
+		t.Run(fmt.Sprintf("ByteSteramOptFinishWrite=%t", flag), func(t *testing.T) {
+			lsID := fmt.Sprintf("logstream%d", i+1)
+			logStreams[lsID] = &logStream{logStreamID: lsID, logicalOffset: 0}
+			ChunkMaxSize(dataSize).Apply(b.client)
+
+			_, err := b.client.WriteBytesWithOptions(b.ctx, lsID, logStreamData, flag)
+			if err != nil {
+				t.Errorf("WriteBytesWithOffset() failed unexpectedly: %v", err)
+			}
+
+			if logStreams[lsID].logicalOffset != int64(dataSize) {
+				t.Errorf("WriteBytesWithOffset() = %d, want %d", logStreams[lsID].logicalOffset, int64(dataSize))
+			}
+			if logStreams[lsID].finalized != bool(flag.(ByteSteramOptFinishWrite)) {
+				t.Errorf("WriteBytesWithOffset() got %t, want %t state", logStreams[lsID].finalized, bool(flag.(ByteSteramOptFinishWrite)))
+			}
+		})
+
+	}
+}
+
+func TestWirteBytesWithOptions_Offset(t *testing.T) {
+	b := setup(t)
+	defer b.shutDown()
+	dataSize := len(logStreamData)
+	for _, offset := range []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteStreamOptOffset(6)} {
+		t.Run(fmt.Sprintf("ByteSteramOptFinishWrite=%t", offset), func(t *testing.T) {
+			lsID := fmt.Sprintf("logstream%d", offset)
+			logStreams[lsID] = &logStream{logStreamID: lsID, logicalOffset: int64(offset.(ByteStreamOptOffset))}
+			ChunkMaxSize(dataSize).Apply(b.client)
+
+			_, err := b.client.WriteBytesWithOptions(b.ctx, lsID, logStreamData[int(offset.(ByteStreamOptOffset)):], offset)
+			if err != nil {
+				t.Errorf("WriteBytesWithOffset() failed unexpectedly: %v", err)
+			}
+			if logStreams[lsID].logicalOffset != int64(dataSize) {
+				t.Errorf("WriteBytesWithOffset() = %d, want %d", logStreams[lsID].logicalOffset, int64(dataSize))
+			}
+			if logStreams[lsID].finalized != true {
+				t.Error("WriteBytesWithOffset() didn't correctly finalize logstream")
+			}
+		})
+	}
+}
+
+type Bytestream struct {
+}
+
+type Server struct {
+	client   *Client
+	listener net.Listener
+	server   *grpc.Server
+	fake     *Bytestream
+	ctx      context.Context
+}
+
+func setup(t *testing.T) *Server {
```

Comment:
`setup` is a generic name. The implementation seems to be doing `newServer` or `newFakeServer`.

Reply:
Done.